### PR TITLE
Adding more CityValidators

### PR DESCRIPTION
**Pull request by sam596**
Working with @jfly, I've created these new validator files with the province names for Argentina, Australia, Brazil, India and the UK, with more to follow. I've collaborated with delegates from the relevant countries to ensure these lists are accurate.

Most of these still need the cityName fields updated in the database to pass the validation.

To-do:

- [ ] Add option for some instances where a province is not required. e.g. "Beijing, China" instead of "Beijing, Beijing, China".
- [ ] Add validation for countries that do not require a province to ensure that no province is listed.
- [ ] Add more countries!

### CODE REVIEW

File: WcaOnRails/lib/country_city_validators/au_city_validator.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  New\ South\ Wales
+  Northern\ Territory
+  Queensland
+  South\ Australia
+  Tasmania
+  Victoria
+  Western\ Australia
+).to_set
+
+CityValidator.add_validator_for_country "AU", CityCommaRegionValidator.new(type_of_region: "state/territory", valid_regions: AU_STATES)
```
Comment (jfly) [inline WcaOnRails/lib/country_city_validators/au_city_validator.rb]: would "state or territory" be nicer here? The error message our users are going to get will look something like "foo is not a valid state/territory" or ""foo is not a valid state or territory".

File: WcaOnRails/lib/country_city_validators/gb_city_validator.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  Aberdeenshire
+  Angus
+  Argyll\ and\ Bute
+  Bedfordshire
+  Berkshire
+  Buckinghamshire
+  Cambridgeshire
+  Cheshire
+  City\ of\ Bristol
+  City\ of\ London
```
Comment (jfly) [inline WcaOnRails/lib/country_city_validators/gb_city_validator.rb]: Does this mean https://www.worldcubeassociation.org/competitions/LondonOpen2017 cityName should change from "London" to "London, City of London"?

**Comment by jfly** [comment]
Also, your TODOs look good to me. Do you want to take care of them in this PR, or in follow up PRs?

**Comment by ronaldmansveld** [comment]
pinging @sam596 for update

**Comment by sam596** [comment]
Apologies, I had very little free time from mid-October to early-November, I should be able to pick this back up later this week.

File: WcaOnRails/lib/country_city_validators/gb_city_validator.rb (hunk 2 above)
Comment (Mollerz) [reply WcaOnRails/lib/country_city_validators/gb_city_validator.rb]: Would change to "Greater London". City of London is a tiny area right in the heart of London that is mostly office space and guild halls...

Comment (sam596) [reply WcaOnRails/lib/country_city_validators/gb_city_validator.rb]: Yeah, I would imagine in the UK, we would change the London 2017 city to something like "Docklands, Greater London"

**Comment by jfly** [comment]
We have a number of tests that create competitions in countries that you've added validators for. Many of those tests are now failing. For instance, [this test](https://github.com/jfly/worldcubeassociation.org/blob/00b66534978cd467eff5c0ecef0e12971deb2bcf/WcaOnRails/spec/requests/media_spec.rb#L61) is attempting to create a competition in the United Kingdom in "San Francisco, California", and it's complaining that "California" is not a valid county. I don't think there's an easy fix here, you'll just have to fix all the broken tests =(

**Comment by sam596** [comment]
So, sorry for the abundance of commits, but this should be good to go.

I now am beginning to get all the updated names for cities with their province ready to edit the existing cityNames when this is merged.

Regarding my to-do list, I have no confidence whatsoever in my Ruby knowledge to be able to code the first two, but with regards to adding more countries, I will continue adding more in other pull requests when I have the information.

**Comment by jfly** [comment]
Awesome work, thanks @sam596!

**Pull request merged**